### PR TITLE
Disallow local byref void methods.

### LINKED ISSUE #8441: Ref returns/locals + Local Methods -- No error on "ref void" local methods
Paste C# code:

```C#
class C
{
    void M()
    {
        ref void NoError() { }
    }

    ref void HasError() { } // CS 8898
}
```

**Expected**: The `ref void` local method `NoError` should be an error, probably with the same error code as the `ref void` member method `HasError`
**Actual**: The `ref void` local method has no errors, and compilation completes successfully

### PR DESCRIPTION
**Pull request by VSadov**
Disallow local byref void methods.

Fixes #8441

### CODE REVIEW

**Comment by VSadov** [comment]
@dotnet/roslyn-compiler - please take a look
Fairly trivial fix.
I am mostly trying to see if branch is ok after merge of byrefs and can accept fixes.

File: src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs [hunk 1]
```diff
@@ -168,6 +168,13 @@ internal TypeSymbol ComputeReturnType()
                 // The return type of an async method must be void, Task or Task<T>
                 diagnostics.Add(ErrorCode.ERR_BadAsyncReturn, this.Locations[0]);
             }
+
+            var returnsVoid = returnType.SpecialType == SpecialType.System_Void;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs]: Consider inlining the local, why calculate this for methods without ref return.

**Comment by AlekseyTs** [comment]
LGTM

File: src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs (hunk 1 above)
Comment (VSadov) [inline src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs]: good point. Thanks

**Pull request merged**